bot/cogs/information.py

```diff
@@ -26,6 +27,51 @@ class Information(Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    @staticmethod
+    def role_can_read(channel: GuildChannel, role: Role) -> bool:
+        """Return True if `role` can read messages in `channel`."""
+        overwrites = channel.overwrites_for(role)
+        return overwrites.read_messages is True
+
+    def get_staff_channel_count(self, guild: Guild) -> int:
+        """
+        Get the number of channels that are staff-only.
+
+        We need to know two things about a channel:
+        - Does the @everyone role have explicit read deny permissions?
+        - Do staff roles have explicit read allow permissions?
+
+        If the answer to both of these questions is yes, it's a staff channel.
+        """
+        channel_ids = set()
+        for channel in guild.channels:
+            if channel.type is ChannelType.category:
+                continue
+
+            if channel in channel_ids:
+                continue  # Only one of the roles has to have read permissions, not all
```

Comment:
You're right, it should be using the id. It's here because The role loop is nested within the channel loop. Once it finds a role with read permissions (which can be determined by checking if the channel has been added to the set), it doesn't need to check the rest of the roles for that channel.

Reply:
OK I see where I went wrong. This should be a break and be inside the role loop.

Reply:
Actually, just break right after adding to the set. Even easier.